test/plots/scatterPlotTests.ts

```diff
@@ -2,395 +2,399 @@
-            assert.strictEqual(selection.attr("fill"), colorAccessor(datum, index, null), "fill is correct");
-          };
-        };
-      };
-
+      let data: any[];
+      let data2: any[];
       beforeEach(() => {
-        svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+        svg = TestMethods.generateSVG();
         xScale = new Plottable.Scales.Linear();
-        xScale.domain([0, 9]);
         yScale = new Plottable.Scales.Linear();
-        yScale.domain([0, 81]);
-        circlePlot = new Plottable.Plots.Scatter<number, number>();
-        circlePlot.addDataset(quadraticDataset);
-        circlePlot.attr("fill", colorAccessor);
-        circlePlot.x((d: any) => d.x, xScale);
-        circlePlot.y((d: any) => d.y, yScale);
-        circlePlot.renderTo(svg);
+        data = [{ x: 0, y: 0 }, { x: 1, y: 1 }];
+        data2 = [{ x: 1, y: 2 }, { x: 3, y: 4 }];
+        dataset = new Plottable.Dataset(data);
+        dataset2 = new Plottable.Dataset(data2);
+        plot = new Plottable.Plots.Scatter<number, number>();
+        plot.x((d: any) => d.x, xScale)
+            .y((d: any) => d.y, yScale)
+            .addDataset(dataset);
       });
 
-      it("setup is handled properly", () => {
-        assert.deepEqual(xScale.range(), [0, SVG_WIDTH], "xScale range was set by the renderer");
-        assert.deepEqual(yScale.range(), [SVG_HEIGHT, 0], "yScale range was set by the renderer");
-        circlePlot.selections().each(getCirclePlotVerifier());
-        assert.strictEqual(circlesInArea, 10, "10 circles were drawn");
+      it("selects all points in all datasets", () => {
+        plot.addDataset(dataset2);
+        plot.renderTo(svg);
+        let allCircles = plot.selections();
+        assert.strictEqual(allCircles.size(), 4, "all circles retrieved");
+        let selectionData = allCircles.data();
+        assert.includeMembers(selectionData, data, "first dataset data in selection data");
+        assert.includeMembers(selectionData, data2, "second dataset data in selection data");
+
         svg.remove();
       });
 
-      it("rendering is idempotent", () => {
-        circlePlot.render();
-        circlePlot.render();
-        circlePlot.selections().each(getCirclePlotVerifier());
-        assert.strictEqual(circlesInArea, 10, "10 circles were drawn");
+      it("selects the closest data point", () => {
+        plot.addDataset(dataset2);
+        plot.renderTo(svg);
+        let closest = plot.entityNearest({
+          x: xScale.scale(0) + 1,
+          y: yScale.scale(0) + 1
+        });
+
+        assert.deepEqual(closest.datum, dataset.data()[0], "correct datum has been retrieved");
+
         svg.remove();
       });
 
-      describe("after the scale has changed", () => {
-        beforeEach(() => {
-          xScale.domain([0, 3]);
-          yScale.domain([0, 9]);
+      it("ignores off-plot data points when retrieving the nearest Entity", () => {
+        plot.addDataset(dataset2);
```

Comment:
Can we make the first dataset more comprehensive such that we do not need a second one?